Added an option to block steam accounts that have not yet setup their profile. Also added config options to edit the kick messages players receive.

Cyanox62 opened the pull request:
Added in config.json
- `"block-non-setup-steam-accounts"` - Bool that can be set in the config just like any other config option. Blocks steam accounts that have not yet been setup.

Added in SCP config
- `vs_no_purchases_message` - Determines the message to be displayed to a user when they are kicked for not having any purchases on their account.
- `vs_non_setup_message` - Determines the message to be displayed to a user when they are kicked for not having their steam profile setup.

EDIT: Added a config validator. This should automatically insert all new configs into existing users' config files.

Bumped version number `3.2.0` -> `3.3.0`

Cyanox62 committed:
Added
-"block-non-setup-steam-accounts"
- vs_no_purchases_message
- vs_non_setup_message

KarlOfDuty commented on VPNShield/VPNShield.cs:
```diff
@@ -63,7 +67,9 @@ public override void Register()
 			this.AddConfig(new ConfigSetting("vs_global", true, true, "Whether or not to use the global config directory, default is true"));
-        }
+			this.AddConfig(new ConfigSetting("vs_no_purchases_message", "This server does not allow new Steam accounts, you have to buy something on Steam before playing.", true, "The message to display to players who are kicked for having a steam account without a purchased game."));
+			this.AddConfig(new ConfigSetting("vs_non_setup_message", "This server does not allow non setup Steam accounts, you have to setup your Steam profile before playing.", true, "The message to display to players who are kicked for having a Steam account that is not yet setup."));
```
These should not be vanilla config options, they should be part of the plugin config.

KarlOfDuty commented on VPNShield/VPNShield.cs:
```diff
@@ -38,12 +38,16 @@ public class VPNShield : Plugin
         public bool autoBlacklistUpdated = false;
 
-        private readonly string defaultConfig =
+		public string noPurchaseKick;
```
This file is inconsistent with tabs/spaces, please convert the file to use space indentation to remove this issue.

KarlOfDuty commented:
Thanks for the PR, I think that could be pretty useful for the plugin! I just need you to change the above before I merge it.

Cyanox62 committed:
Fixed space consistency, changed vanilla configs to use json config

Cyanox62 committed:
Fixed spacing again

Cyanox62 committed:
Added a config validator

Cyanox62 commented:
I've changed the configs to be in the json file instead and fixed the spacing. I also added a config validator so that players updating their plugin should just be able to drag and drop and it'll fix their config with the new options automatically. So you can disregard what I said above about having to manually add that new line.

KarlOfDuty commented:
Sounds good, but you still need to fix the whitespace issues
![image](https://user-images.githubusercontent.com/18099825/67798096-35b19100-fa83-11e9-83fb-e2d7306a815b.png)

KarlOfDuty reviewed (changes requested):
As per previous comment, the whitespace is messed up.

Cyanox62 committed:
Fixed spacing
